src/components/Roles/Roles.js

```diff
@@ -29,27 +33,45 @@ class Roles extends Component {
     });
   }
 
-  static navigatorStyle = {
-    navBarHidden: true,
-  };
-
-  // This method is invoked right before calling the render method, both on the initial mount and on subsequent updates.
-  // We use it to detect if the role has changed to make the transition to the next screen.
-  static getDerivedStateFromProps(nextProps) {
-    if (nextProps.role !== null) {
-      Application.startLoggedInApp(nextProps.role);
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.role) {
+      switch (nextProps.role) {
+        case strings.gatherAction:
+          // this.props.navigator.push({
+          //  screen: Screens.Gather,
+          //  animationType: 'fade',
+          // });
+          Application.startRole(nextProps.role);
+          break;
+        case strings.weighAction:
+          // this.props.navigator.push({
+          //  screen: Screens.Weigh,
+          //  animationType: 'fade',
+          // });
+          Application.startRole(nextProps.role);
+          break;
+        case strings.baleAction:
+          // this.props.navigator.push({
+          //  screen: Screens.Bale,
+          //  animationType: 'fade',
+          // });
+          Application.startRole(nextProps.role);
+          break;
+        default:
+          break;
+      }
```

Comment:
Commented code will be the one used in the future when the new header is merged into master